src/style/style_layer/symbol_style_layer.js

```diff
@@ -77,6 +98,83 @@ class SymbolStyleLayer extends StyleLayer {
+            if (!SymbolStyleLayer.hasPaintOverride(this.layout, overridable)) {
+                continue;
+            }
+            const overriden = this.paint.get(overridable);
+            const override = new FormatSectionOverride(overriden);
+            const styleExpression = new StyleExpression(override, overriden.property.specification);
+            let expression = null;
+            if (overriden.value.kind === 'constant' || overriden.value.kind === 'source') {
+                expression = (new ZoomConstantExpression('source', styleExpression): SourceExpression);
+            } else {
+                expression = (new ZoomDependentExpression('composite',
+                                                          styleExpression,
+                                                          overriden.value.zoomStops,
+                                                          overriden.value._interpolationType): CompositeExpression);
+            }
+            this.paint._values[overridable] = new PossiblyEvaluatedPropertyValue(overriden.property,
+                                                                                 expression,
+                                                                                 overriden.parameters);
+        }
+    }
+
+    _handleOverridablePaintPropertyUpdate<T, R>(name: string, oldValue: PropertyValue<T, R>, newValue: PropertyValue<T, R>): boolean {
+        if (!this.layout || oldValue.isDataDriven() || newValue.isDataDriven()) {
+            return false;
+        }
+        return SymbolStyleLayer.hasPaintOverride(this.layout, name);
+    }
+
+    static hasPaintOverride(layout: PossiblyEvaluated<LayoutProps>, propertyName: string): boolean {
+        const textField = layout.get('text-field');
+
+        let sections: any = [];
+        if (textField.value.kind === 'constant' && textField.value.value instanceof Formatted) {
+            sections = textField.value.value.sections;
+        } else if (textField.value.kind === 'source') {
+            const checkExpression = (expression: Expression) => {
+                if (sections.length) {
+                    return;
+                }
+
+                if (expression instanceof Literal && typeOf(expression.value) === FormattedType) {
+                    const formatted: Formatted = ((expression.value): any);
+                    sections = formatted.sections;
+                } else if (expression instanceof FormatExpression) {
+                    sections = expression.sections;
+                }
+            };
+
+            const expr: ZoomConstantExpression<'source'> = ((textField.value): any);
+            if (expr._styleExpression) {
+                checkExpression(expr._styleExpression.expression);
+                if (!sections.length) {
+                    expr._styleExpression.expression.eachChild(checkExpression);
+                }
+            }
+        }
+
```

Comment:
I think at this point `sections` is set to the sections of a single formatted expression. In the case where there is a ZoomConstantExpression with multiple possible formatted expressions it will only hold the last one. If the last expression has no overrides but an earlier one does, does this return the wrong value?

For example:

```json
["case",
    ["get", "colourful"], ["format", "red text", { "text-color": "red" }],
    ["format", "default text", {}]]
```